### PR TITLE
Deep-merge Select2 options

### PR DESCRIPTION
**Pull request by michalkurzeja**
In order to be able to provide custom options for the nested keys (e.g. `ajax.data`) the options must be recursively merged. Otherwise all other sub-keys must also be provided, or copied.

E.g. if you want to provide custom parameters to the query you can pass such an options object when initialising the `select2entity` plugin:

```
{
    ajax: {
        data: function() {
            // some implementation...
        }
    }
}
```

This will override the custom `transport` and `processResults` methods provided by this bundle and will result in some functionality loss (infinite scroll and cache will not work anymore).

### CODE REVIEW

**Comment by tetranz** [comment]
Thanks. It's about time I tagged a new release. I'll do that soon.

**Comment by michalkurzeja** [comment]
Cool, thanks!
